registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/UserNotificationApi.java

```diff
@@ -0,0 +1,316 @@
+				buildResponse(ctx, description.getMessage(), null);
+				regProcLogger.debug("UserNotificationApi:processURL called ended for registration id {}", rid);
+
+		} catch (UserNotificationRequestValidationException e) {
+			logError(rid, e.getErrorCode(), e.getMessage(), e, ctx);
+		} catch (UserNotificationException e) {
+			logError(rid, e.getErrorCode(), e.getMessage(), e, ctx);
+		} catch (Exception e) {
+			logError(rid, PlatformErrorMessages.RPR_UNA_UNKNOWN_EXCEPTION.getCode(),
+					PlatformErrorMessages.RPR_UNA_UNKNOWN_EXCEPTION.getMessage(), e, ctx);
+		}
+
+	}
+
+	/**
+	 * @param routingContext
+	 */
+	private void failure(RoutingContext routingContext) {
+		this.setResponse(routingContext, routingContext.failure().getMessage());
+	}
+
+	/**
+	 *
+	 */
+	@Override
+	public MessageDTO process(MessageDTO object) {
+		return null;
+	}
+
+	/**
+	 * @param ctx
+	 * @return
+	 */
+	private String getUser(RoutingContext ctx) {
+		String user = "";
+		if (Objects.nonNull(ctx.user()) && Objects.nonNull(ctx.user().principal()))
+			user = ctx.user().principal().getString("username");
+		return user;
+	}
+
+	/**
+	 * Update audit.
+	 *
+	 * @param description             the description
+	 * @param registrationId          the registration id
+	 * @param isTransactionSuccessful the is transaction successful
+	 */
+	private void updateAudit(LogDescription description, String registrationId, boolean isTransactionSuccessful,
+			String user) {
+
+		String moduleId = isTransactionSuccessful ? MODULE_ID : description.getCode();
+
+		String eventId = isTransactionSuccessful ? EventId.RPR_402.toString() : EventId.RPR_405.toString();
+		String eventName = isTransactionSuccessful ? EventName.UPDATE.toString() : EventName.EXCEPTION.toString();
+		String eventType = isTransactionSuccessful ? EventType.BUSINESS.toString() : EventType.SYSTEM.toString();
+
+		auditLogRequestBuilder.createAuditRequestBuilder(description.getMessage(), eventId,
+				eventName, eventType,
+				moduleId, MODULE_NAME, registrationId, user);
+	}
+
+
+
+	/**
+	 * @param routingContext
+	 * @param message
+	 * @param errors
+	 */
+	private void buildResponse(RoutingContext routingContext, String message, List<ErrorDTO> errors) {
+		UserNotificationResponse UserNotificationResponse = new UserNotificationResponse();
+		UserNotificationResponse.setId(id);
+		UserNotificationResponse.setVersion(version);
+		UserNotificationResponse.setResponsetime(DateUtils.getUTCCurrentDateTimeString(dateTimePattern));
+		if (message == null) {
+			UserNotificationResponse.setErrors(errors);
+		} else {
+			ResponseDTO responseDTO = new ResponseDTO();
+			responseDTO.setStatusMessage(message);
+			UserNotificationResponse.setResponse(responseDTO);
+		}
+		this.setResponse(routingContext, UserNotificationResponse);
+
+	}
+
+	/**
+	 * @param rid
+	 * @param errorCode
+	 * @param errorMessage
+	 * @param e
+	 * @param ctx
+	 */
+	private void logError(String rid, String errorCode, String errorMessage,
```

Comment:
This method is not only logging error, but also resending back an error response, so name should convey the both

Reply:
Method name changed to "logErrorAndSendNotifivation"

Reply:
notification - spell check

Reply:
Done